tekdoc/main.go

```diff
@@ -17,15 +17,67 @@ limitations under the License.
 package main
 
 import (
-	"fmt"
+	"flag"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/scheme"
+	"io/ioutil"
+	"log"
+	"os"
+	"text/template"
+)
 
-	_ "github.com/tektoncd/pipeline/pkg/pod"
+var (
+	filename = flag.String("f", "", "Name of the file to parse")
 )
 
+type a struct {
+	Taskname  string
+	Input     []v1alpha1.ParamSpec
+	Resources []v1alpha1.TaskResource
+	Output    []v1alpha1.TaskResource
+}
+
 func main() {
-	fmt.Println("Hello this is STILL TekDoc")
 
 	// TODO: take a file path, parse it as YAML -> v1alpha1.Task
+	flag.Parse()
+	dat, err := ioutil.ReadFile(*filename)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var task v1alpha1.Task
+
+	if _, _, err := scheme.Codecs.UniversalDeserializer().Decode(dat, nil, &task); err != nil {
+		log.Fatal(err)
+	}
+
 	// TODO: use Go templates to print v1alpha1.Task as Markdown
-	// TODO: tests!
+
+	x := a{
+		Taskname:  task.Name,
+		Input:     task.Spec.Inputs.Params,
+		Resources: task.Spec.Inputs.Resources,
+		Output:    task.Spec.Outputs.Resources,
+	}
+
+	t := `# {{.Taskname}}
+## Install the Task \n kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/{{.Taskname}}/{{.Taskname}}.yaml
+### Input:-
+{{range .Input}}- {{.Name}}, {{.Description}}
+{{end}}
+### Resources:-
+{{range .Resources}}- {{.ResourceDeclaration.Name}}, {{.ResourceDeclaration.Type}}
+{{end}}
+### Output:-
+{{range .Output}}- {{.ResourceDeclaration.Name}}, {{.ResourceDeclaration.Type}}
+{{end}}`
+
+	tmpl := template.Must(template.New("test").Parse(t))
+	err = tmpl.Execute(os.Stdout, x)
```

Comment:
This is fine, but a good habit to get into is to prefer using:
```go
if err := ...; err != nil {
 ...
}
```

whereever possible. This helps avoid problems of not checking errors / overwriting values.

Reply:
Noted!